main.cc

```diff
@@ -1071,9 +1071,14 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 reinterpret_cast<service::storage_proxy_stats::stats*>(ptr)->register_stats();
                 reinterpret_cast<service::storage_proxy_stats::stats*>(ptr)->register_split_metrics_local();
             };
+            auto desc_ring_func = sharded_parameter([&] {
+                return service::storage_proxy::describe_ring_func([ss = std::ref(ss)] (const sstring& keyspace, bool include_only_local_dc) {
+                    return ss.get().local().describe_ring(keyspace, include_only_local_dc);
+                });
+            });
```

Comment:
Please, no. Yes, in order to detach storage_service::describe_ring() from storage_service it takes efforts and time, but it should be done.

The describe_ring() and everything it calls below (construct_range_to_endpoint_map, get_range_to_address_map, get_range_to_address_map_in_local_dc, get_tokens_in_local_dc, ...) can live without storage_service. I'd suggest moving them somewhere into locator/. The storage_service::get_rpc_address() cannot just become standalone static helper, but it looks OK to make it gossiper method. Both storage_service and _proxy have gossiper dependency to call it on.

Reply:
I'd be happy to merge the PR detaching describe_ring() from storage_service